rts/src/main/java/eta/runtime/Runtime.java

```diff
@@ -157,8 +162,21 @@ public static boolean debugMemoryManager() {
         return debugMemoryManager;
     }
 
+    public static void init_params() {
+        RuntimeOptions rto = new RuntimeOptions(RTS_PROPERTIES_PATH);
+        // Initialize parameters explicitly
+        maxWorkerCapabilities = rto.getInt("maxWorkerCapabilities", getNumberOfProcessors());
+        maxGlobalSparks = rto.getInt("maxGlobalSparks", 4096);
+        minTSOIdleTime = rto.getInt("minTSOIdleTime", 20);
+        maxTSOBlockTime = rto.getInt("maxTSOBlockTime", 1);
+        minWorkerCapabilityIdleTime = rto.getInt("minWorkerCapabilityIdleTime", 1000);
+        gcOnWeakPtrFinalization = rto.getBoolean("gcOnWeakPtrFinalization", false);
+        maxLocalSparks = rto.getInt("maxLocalSparks", 4096);
+    }
+
     public static void main(String[] args, Closure mainClosure) throws Exception {
         Runtime.setProgramArguments(args);
+        init_params();
```

Comment:
Foreign exports (calling Eta from Java) doesn't always call `Runtime.main` so the parameters won't get initialized in that case. Instead, place the call to parameter initialization in a static initializer for this class.